docs/rfc/0003_risingwave_ng.md

```diff
@@ -0,0 +1,208 @@
+  + Declare complex groups of pods of components, specifying different templates, images, resources and other things related
+  + Specifying the listen ports of components
+  + The configs in one `group` should be merged into the globals, e.g.,
+    + maps are merged and overridden if there're new values (resources)
+    + lists are merged
+    + values are overridden
+  + The groups here are extended groups to the global ones, i.e. if there're 2 compactors in global and a group of 2 compactors in components, there will be 4 in total
++ `configuration` for using pre-defined configs
++ `storages` for specifying the storage types and credentials, as well as the PVC template for compute nodes
++ `security` for specifying the TLS configs
+
+Also, the `podTemplate` will be read and applied onto the Pods of RisingWave if it is set.
+
+Here're examples to demonstrate the new CRD:
+
++ 1/1/1/1, 1C1G
+
+```yaml
+apiVersion: risingwave.singularity-data.com/v1alpha1
+kind: RisingWave
+metadata:
+  name: test-risingwave
+spec:
+  global:
+    image: ghcr.io/singularity-data/risingwave:latest
+    replicas:
+      meta: 1
+      frontend: 1
+      compute: 1
+      compactor: 1
+    resources:
+      limits:
+        cpu: 1
+        memory: 1Gi
+  storages:
+    meta:
+      memory: true
+    object:
+      memory: true
+```
+
++ 1/1/1/1, 2 spot compactors on a node group labeled with "spot", 1C1G
+
+```yaml
+apiVersion: risingwave.singularity-data.com/v1alpha1
+kind: RisingWave
+metadata:
+  name: test-risingwave
+spec:
+  # ...
+  components:
+    compactor:
+      groups:
+      - name: spot
+        replicas: 2
+        nodeSelector:
+          node-group: spot
+```
+
++ Setting the Pods to be privileged
+
+```yaml
+apiVersion: risingwave.singularity-data.com/v1alpha1
+kind: RisingWavePodTemplate
+metadata:
+  name: privileged-pods
+spec:
+  spec:
+    containers:
+    - securityContext:
+        privileged: true
+---
+apiVersion: risingwave.singularity-data.com/v1alpha1
+kind: RisingWave
+metadata:
+  name: test-risingwave
+spec:
+  global:
+    podTemplate: privileged-pods
+```
+
++ Mixed arch instance (compactors, explicitly)
+
+```yaml
+apiVersion: risingwave.singularity-data.com/v1alpha1
+kind: RisingWave
+metadata:
+  name: test-risingwave
+spec:
+  global:
+    image: ghcr.io/singularity-data/risingwave:latest
+    replicas:
+      meta: 1
+      frontend: 1
+      compute: 1
+    resources:
+      limits:
+        cpu: 1
+        memory: 1Gi
+    nodeSelector:
+      kubernetes.io/arch: amd64
+  components:
+    compactor:
+      groups:
+      - name: group-amd64
+        replicas: 1
+        image: ghcr.io/singularity-data/risingwave:latest
+        nodeSelector:
+          kubernetes.io/arch: amd64
+      - name: group-arm64
+        image: public.ecr.aws/x5u3w5h6/risingwave-arm:latest
+        replicas: 1
+        nodeSelector:
+          kubernetes.io/arch: arm64
+```
+
++ Restart the frontend Pods (update)
```

Comment:
Is this how to restart the pods? Update its `restartAt` spec?

Reply:
> Is this how to restart the pods? Update its `restartAt` spec?

Right. This is inspired by how they restart the Pods in workloads like Deployment/StatefulSet, by setting an annotation of restarting time on the Pod to trigger re-creating of the Pods. And I'm going to implement in the same way.

Reply:
Looks good!